packages/discord.js/typings/index.d.ts

```diff
@@ -4669,6 +4730,23 @@ export interface SelectMenuComponentOptionData {
   value: string;
 }
 
+export interface TextInputComponentData extends BaseComponentData {
+  customId: string;
+  style: TextInputStyle;
+  label: string;
+  minLength?: number;
+  maxLength?: number;
+  required?: boolean;
+  value?: string;
+  placeholder?: string;
+}
+
+export interface ModalData {
+  customId: string;
+  title: string;
+  components: ActionRowData<ModalActionRowComponentData>;
```

Comment:
```suggestion
  custom_id: string;
  title: string;
  components: ActionRowData<ModalActionRowComponentData>[];
```

these are the correct structures to keep typescript happy while directly inputting an object into the Modal constructor

Reply:
This is incorrect we use `APIModalCallbackResponseData` for snake cased keys.